### PR TITLE
add Aac audio video streaming sample

### PR DESCRIPTION
**Pull request by chehefen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: samples/KvsAacAudioVideoStreamingSample.c [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    CHK_STATUS(createDefaultDeviceInfo(&pDeviceInfo));
+    // adjust members of pDeviceInfo here if needed
+    pDeviceInfo->clientInfo.loggerLogLevel = LOG_LEVEL_DEBUG;
+
+    CHK_STATUS(createRealtimeAudioVideoStreamInfoProvider(streamName, DEFAULT_RETENTION_PERIOD, DEFAULT_BUFFER_DURATION, &pStreamInfo));
+
+    // adjust members of pStreamInfo here if needed
+
+    // set up audio cpd.
+    for(UINT32 i = 0; i < 2; ++i) {
```
Comment (MushMal) [inline samples/KvsAacAudioVideoStreamingSample.c]: LOL - you could simply do one after the other

File: samples/KvsAacAudioVideoStreamingSample.c [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+    for(i = 0; i < NUMBER_OF_AAC_FRAME_FILES; ++i) {
+        SAFE_MEMFREE(data.audioFrames[i].buffer);
+    }
+
+    for(i = 0; i < NUMBER_OF_H264_FRAME_FILES; ++i) {
+        SAFE_MEMFREE(data.videoFrames[i].buffer);
+    }
+
+    if (pDeviceInfo != NULL) {
```
Comment (MushMal) [inline samples/KvsAacAudioVideoStreamingSample.c]: For all of these cases, just do freeXXX - those are idempotent. Just make sure the pBlah is initialized to NULL first

File: samples/KvsAacAudioVideoStreamingSample.c [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    for(i = 0; i < NUMBER_OF_AAC_FRAME_FILES; ++i) {
+        SNPRINTF(filePath, MAX_PATH_LEN, "%s/aacSampleFrames/sample-%03d.aac", data.sampleDir, i + 1);
+        CHK_STATUS(readFile(filePath, TRUE, NULL, &fileSize));
+        data.audioFrames[i].buffer = (PBYTE) MEMALLOC(fileSize);
+        data.audioFrames[i].size = fileSize;
+        CHK_STATUS(readFile(filePath, TRUE, data.audioFrames[i].buffer, &fileSize));
+    }
+    printf("Done loading audio frames.");
+
+    printf("Loading video frames...");
```
[the remark this answers is not in the document]
Comment (chehefen) [reply samples/KvsAacAudioVideoStreamingSample.c]: I think we should try to not have any pic macros in the sample.

[the remark this answers is not in the document]
Comment (MushMal) [reply samples/KvsAacAudioVideoStreamingSample.c]: We have discussed some of this. It's OK in my opinion to not use some of the code-flow macros like CHK_STATUS. It's also OK to get rid of DLOG/defaultLog but I would like to see consistent usage of printf then.

File: samples/KvsAacAudioVideoStreamingSample.c [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+    frame.index = 0;
+
+    // video track is used to mark new fragment. A new fragment is generated for every frame with FRAME_FLAG_KEY_FRAME
+    frame.flags = fileIndex % DEFAULT_KEY_FRAME_INTERVAL == 0 ? FRAME_FLAG_KEY_FRAME : FRAME_FLAG_NONE;
+
+    while (defaultGetTime() < data->streamStopTime) {
+        status = putKinesisVideoFrame(data->streamHandle, &frame);
+        ATOMIC_STORE_BOOL(&data->firstVideoFramePut, TRUE);
+        if (STATUS_FAILED(status)) {
+            DLOGD("putKinesisVideoFrame failed with 0x%08x", status);
```
Comment (MushMal) [inline samples/KvsAacAudioVideoStreamingSample.c]: Do you want to change this to printf?

File: samples/KvsAacAudioVideoStreamingSample.c [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+    frame.presentationTs = 0; // relative time mode
+    frame.index = 0;
+    frame.flags = FRAME_FLAG_NONE; // audio track is not used to cut fragment
+
+    while (defaultGetTime() < data->streamStopTime) {
+        // no audio can be put until first video frame is put
+        if (ATOMIC_LOAD_BOOL(&data->firstVideoFramePut)) {
+            status = putKinesisVideoFrame(data->streamHandle, &frame);
+            if (STATUS_FAILED(status)) {
+                DLOGD("putKinesisVideoFrame for audio failed with 0x%08x", status);
```
Comment (MushMal) [inline samples/KvsAacAudioVideoStreamingSample.c]: same

File: samples/KvsAacAudioVideoStreamingSample.c [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+    MEMSET(&data, 0x00, SIZEOF(SampleCustomData));
+
+    /*
+     * user is expected to set audio cpd for their stream.
+     * 5 bits (Audio Object Type) | 4 bits (frequency index) | 4 bits (channel configuration) | 3 bits (not used)
+     */
+    BYTE audioCpd[] = {0x11, 0x90, 0x56, 0xe5, 0x00};
+
+    if (argc < 2) {
+        defaultLogPrint(LOG_LEVEL_ERROR, "", "Usage: AWS_ACCESS_KEY_ID=SAMPLEKEY AWS_SECRET_ACCESS_KEY=SAMPLESECRET %s <stream_name> <duration_in_seconds> <frame_files_path>\n", argv[0]);
```
Comment (MushMal) [inline samples/KvsAacAudioVideoStreamingSample.c]: printf?

**Pull request merged**